src/vivarium_nih_us_cvd/components/risks.py

```diff
@@ -95,3 +126,55 @@ def _get_current_exposure(self, index: pd.Index) -> pd.Series:
         return (
             self.gbd_exposure(index) * self.population_view.get(index)[COLUMNS.SBP_MULTIPLIER]
         )
+
+    def _get_target_modifier(
+        self, builder: Builder
+    ) -> Callable[[pd.Index, pd.Series], pd.Series]:
+        """Apply medication effects"""
+
+        def adjust_target(index: pd.Index, target: pd.Series) -> pd.Series:
+            """Determine the exposure decrease as treatment_efficacy * adherence_score"""
+            pop_view = self.population_view.get(index)
+            adherence_scores = pop_view[COLUMNS.SBP_MEDICATION_ADHERENCE].map(
+                MEDICATION_ADHERENCE_SCORE
+            )
+            df_efficacy = pd.DataFrame(
+                {"bin": pd.cut(x=target, bins=sbp_bin_edges, right=True)}
+            )
+            df_efficacy["sbp_start_exclusive"] = df_efficacy["bin"].apply(lambda x: x.left)
```

Comment:
I'd like to talk about this implementation. I think there's a simpler way to do this, but it'd be easier to work through it together

Reply:
Happy to chat about if you decide it's a priority